pycbc/events/status.py

```diff
@@ -0,0 +1,252 @@
+        ### LIGO / Virgo operation!!!!
+        if (('_HW_INJ' in segment_name and not 'NO' in segment_name) or 
+           'VETO' in segment_name):
+            data = query_flag(ifo, 'DATA', start_time, end_time)
+             
+            if '_HW_INJ' in segment_name:
+                name = 'NO_' + segment_name    
+            else:
+                name = segment_name.replace('_VETO', '')
+
+            negate = query_flag(ifo, name, start_time, end_time)
+            return (data - negate).coalesce()
+
+        duration = end_time - start_time
+        url = 'https://www.gw-openscience.org/timeline/segments/json/O1/{}_{}/{}/{}/'
+        url = url.format(ifo, segment_name, start_time, duration)
+        
+        try:
+            flag_segments = json.loads(urllib.urlopen(url).read())['segments']
+        except:
+            msg = "Unable to find segments in GWOSC, check flag name or times"
+            if source == 'any':
+                return query_flag(ifo, segment_name, start_time, end_time,
+                                  source=='dqsegdb')
+            else:
+                raise ValueError(msg)
+    elif source == 'dqsegdb':
+        # Let's not hard require dqsegdb to be installed if we never get here.
+        try:
+            from dqsegdb.apicalls import dqsegdbQueryTimes as query
+        except ImportError:
+            raise ValueError("Could not query flag. DQSegdb is not installed" 
+                             "to additionally try there")
+
+        # The veto definer will allow the use of MACRO names
+        # These directly correspond the name defined in the veto definer file.
+        if veto_definer is not None:
+            veto_def = parse_veto_definer(veto_definer)
+
+        # We treat the veto definer name as if it were its own flag and
+        # a process the flags in the veto definer
+        flag_segments = segmentlist([])
+        if veto_definer is not None and segment_name in veto_def[ifo]:
+            for flag in veto_def[ifo][segment_name]:
+                segs = query("https", server, ifo, flag['name'],
+                             flag['version'], 'active',
+                             start_time, end_time)[0]['active']
+                for rseg in segs:
+                    s, e = rseg[0] + flag['start_pad'], rseg[1] + flag['end_pad']
+                    flag_segments.append(segment(s, e))
+        else: # Standard case just query directly.
+            try:
+                name, version = segment_name.split(':')
+                segs = query("https", server, ifo, name, version, 
+                      'active', start_time, end_time)[0]['active']
+                for rseg in segs:
+                    s, e = rseg[0], rseg[1]
+                    flag_segments.append(segment(s, e))
+            except:
+                raise ValueError("Could not query flag, check name (%s) or times",
+                                 segment_name)
+
```

Comment:
I ended up here having apparently not gone down any of these if branches ... Need an else: Raise Error here

Reply:
I can't match up your comment with the line. Do you mean you got this exception? Or do you mean something else?